src/System.Private.CoreLib/shared/System/Diagnostics/Debug.cs

```diff
@@ -6,15 +6,42 @@
 #define DEBUG
 using System.Diagnostics.Contracts;
 using System.Runtime.CompilerServices;
+using System.Threading;
 
 namespace System.Diagnostics
 {
+    public abstract class DebugProvider
+    {
+        public virtual bool AssertUIEnabled { get { return false; } set { } }
+        public virtual void ShowDialog(string stackTrace, string message, string detailMessage, string errorSource) { }
+        public abstract void Write(string message);
+    }
+
     /// <summary>
     /// Provides a set of properties and methods for debugging code.
     /// </summary>
     public static partial class Debug
     {
         private static readonly object s_lock = new object();
+        private static volatile DebugProvider s_provider;
```

Comment:
I think you can just create the DebugInternal directly here, like:

`private static DebugProvider s_provider = new DebugInternal();`